src/menu.rs

```diff
@@ -0,0 +1,195 @@
+#[derive(Debug, Clone)]
+pub struct Menu {
+  pub title: String,
+  pub items: Vec<MenuItem>,
+}
+
+impl Menu {
+  pub fn new(title: &str, items: Vec<MenuItem>) -> Self {
+    Self {
+      title: String::from(title),
+      items,
+    }
+  }
+}
+
+#[derive(Debug, Clone)]
+/// CustomMenu is a custom menu who emit an event inside the EventLoop.
+pub struct CustomMenu {
+  // todo: replace id by a type that
+  // we can send from the with_menu::<T>()
+  // or we could check to use the UserEvent::<T> but
+  // my latest test failed
+  pub id: String,
+  pub name: String,
+  pub keyboard_accelerators: Option<&'static str>,
+}
+
+/// A menu item, binded to a pre-defined action or `Custom` emit an event.
+#[derive(Debug, Clone)]
+pub enum MenuItem {
+  /// A custom menu emit an event inside the EventLoop.
+  Custom(CustomMenu),
+
+  /// Shows a standard "About" item
+  ///
+  /// ## Platform-specific
+  ///
+  /// - **Windows/Linux:** Unsupported (noop).
```

Comment:
Lets list them all lol
```suggestion
  /// - **Windows / Linux / Android / iOS:** Unsupported (noop).
```